pkg/controller/direct/logging/logmetric_controller.go

```diff
@@ -228,66 +228,110 @@ func logMetricStatusToKRM(in *api.LogMetric, out *krm.LoggingLogMetricStatus) er
-	}
+		if ValueOf(a.desired.Spec.ValueExtractor) != a.actual.ValueExtractor {
+			update.ValueExtractor = ValueOf(a.desired.Spec.ValueExtractor)
+		}
+		if a.desired.Spec.LoggingLogBucketRef != nil && a.desired.Spec.LoggingLogBucketRef.External != a.actual.BucketName {
+			update.BucketName = a.desired.Spec.LoggingLogBucketRef.External
+		}
 
-	// DANGER: this is an upsert; it will create the LogMetric if it doesn't exists
-	// but this behavior is consistent with the DCL backed behavior we provide for this resource.
-	// todo acpana: look for / switch to a better method and/or use etags etc
-	actualUpdate, err := a.logMetricClient.Update(a.fullyQualifiedName(), update).Context(ctx).Do()
-	if err != nil {
-		return fmt.Errorf("logMetric update failed: %w", err)
+		// DANGER: this is an upsert; it will create the LogMetric if it doesn't exists
+		// but this behavior is consistent with the DCL backed behavior we provide for this resource.
+		// todo acpana: look for / switch to a better method and/or use etags etc
+		updated, err := a.logMetricClient.Update(a.fullyQualifiedName(), update).Context(ctx).Do()
+		if err != nil {
+			return fmt.Errorf("logMetric update failed: %w", err)
+		}
+		latest = updated
 	}
 
 	status := &krm.LoggingLogMetricStatus{}
-	if err := logMetricStatusToKRM(actualUpdate, status); err != nil {
+	if err := logMetricStatusToKRM(latest, status); err != nil {
 		return err
 	}
 
 	// actualUpdate may not contain the description for the metric descriptor.
-	if update.Description != "" {
+	if latest.Description != "" {
 		if status.MetricDescriptor != nil {
-			status.MetricDescriptor.Description = &update.Description
+			status.MetricDescriptor.Description = &latest.Description
 		}
 	}
 
 	return setStatus(u, status)
 }
 
+func (a *logMetricAdapter) hasChanges(ctx context.Context, u *unstructured.Unstructured) bool {
+	log := klog.FromContext(ctx)
+
+	if u.GetGeneration() != getObservedGeneration(u) {
+		log.V(2).Info("generation does not match", "generation", u.GetGeneration(), "observedGeneration", getObservedGeneration(u))
+		return true
+	}
+
+	gcpUpdateTime := a.actual.UpdateTime
+	if gcpUpdateTime == "" {
+		log.Info("updateTime is not set in GCP")
+		return true
+	}
+	gcpUpdateTimestamp, err := time.Parse(time.RFC3339, gcpUpdateTime)
+	if err != nil {
+		log.Error(err, "gcp updateTime cannot be parsed", "updateTime", gcpUpdateTime)
+		return true
+	}
+
+	obj := &krm.LoggingLogMetric{}
+	if err := runtime.DefaultUnstructuredConverter.FromUnstructured(u.Object, &obj); err != nil {
+		log.Error(err, "error converting from unstructured")
+		return true
+	}
+	if obj.Status.UpdateTime == nil {
+		log.Info("status.updateTime is not set")
+		return true
+	}
+
+	objUpdateTimestamp := obj.Status.UpdateTime.Time
+	if objUpdateTimestamp == gcpUpdateTimestamp {
+		log.V(2).Info("status.updateTime matches gcp updateTime", "status.updateTime", ValueOf(obj.Status.UpdateTime), "gcpUpdateTime", gcpUpdateTime)
+		return false
+	}
```

Comment:
It would be nice to see if we can see reduced requests in our http golden logs with a step scenario where we catch spurious updates.

More to the point, we should see how this does against our test for mutable but unreadable fields